lms/djangoapps/course_api/serializers.py

```diff
@@ -113,6 +113,7 @@ class CourseSerializer(serializers.Serializer):  # pylint: disable=abstract-meth
     mobile_available = serializers.BooleanField()
     hidden = serializers.SerializerMethodField()
     invitation_only = serializers.BooleanField()
+    certificate_available_date = serializers.DateTimeField()
```

Comment:
I think we can improve this a little bit, Currently `certificate_available_date` is returned no matter if the certificate display behavior is enabled or not. Although this might be more like a business decision I'd mention what I think here should be done here:

1. I think the addition of this field here in the course details API should be conditional on whether to include this date in the API or not.
2. The conditions that I propose are that before adding this to the response of course details API we should check if the certificate display behavior is enabled or not. 
3. We should only include `certificate_available_date` in the API if the `certificate_display_behaviour` is enabled otherwise this can be None.

For point#3, _How do we enable this behavior?_ I've mentioned it in https://github.com/mitodl/mitxonline/issues/1339#issue-1530615721.

As a reference, I looked at the implementation of other APIs (e.g. [Courseware API](https://courses.edx.org/api-docs/#/courseware/courseware_course_read)) that include this date and they also check for the certificate display behavior to be enabled before adding the `certificate_available_date` into the API response.

[Here](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/certificates/api.py#L299-L311) is how it's done for the courseware API in the code.

@pdpinch What are your thoughts on this?

Reply:
I agree, particularly because the other APIs check this course setting as well.

Reply:
@pdpinch Do we need `certificate_available_date` in the [CourseListAPI](https://courses.edx.org/api-docs/#/courses/courses_v1_courses_list) and [CourseDetailAPI](https://courses.edx.org/api-docs/#/courses/courses_v1_courses_read) or only in [CourseDetailAPI](https://courses.edx.org/api-docs/#/courses/courses_v1_courses_read)?

Reply:
I don't know how [CourseListAPI](https://courses.edx.org/api-docs/#/courses/courses_v1_courses_list) is used, but I would prefer that we don't expand the scope of this PR.